rero_ils/modules/holdings/api.py

```diff
@@ -261,28 +262,24 @@ def _get_next_issue_display_text(cls, patterns):
         :param patterns: List of a valid holdings patterns.
         :returns: A display text of the next predicted issue.
         """
-        texts = []
-        for pattern in patterns:
-            text = []
+        issue_data = {}
+        for pattern in patterns.get('values', []):
             levels = pattern.get('levels', [])
+            pattern_name = pattern.get('name', '')
+            level_data = {}
             for level in levels:
-                show_level = level.get('show', True)
-                if show_level:
-                    text_value = level.get('next_value', level.get(
-                        'starting_value', 1))
-                    mapping = level.get('mapping_values')
-                    if mapping:
-                        text_value = mapping[text_value - 1]
-                    text.append(
-                        ' '.join(
-                            (level.get('name', ''), str(text_value))
-                        ).lstrip())
-            texts.append(' '.join(text))
-        return ' '.join(texts)
+                text_value = level.get('next_value', level.get(
+                    'starting_value', 1))
+                mapping = level.get('mapping_values')
+                if mapping:
+                    text_value = mapping[text_value - 1]
+                level_data.update({level.get('name', ''): str(text_value)})
+            issue_data[pattern_name] = level_data
+        return Template(patterns.get('template')).render(**issue_data)
 
     def increment_next_prediction(self):
         """Increment next prediction."""
-        if not self.patterns:
+        if not self.patterns or not self.patterns.get('values'):
```

Comment:
`or` ? are you sure that's not an `and` operator that should be used ?

Reply:
Yes it is `or`. because I need to check if one of the keys is missing